src/hera/workflows/_mixins.py

```diff
@@ -616,28 +616,13 @@ class CallableTemplateMixin(ArgumentsMixin):
     not duplicated.
     """
 
-    def __call__(self, *args, **kwargs) -> Union[None, Step, Task]:
-        if "name" not in kwargs:
-            kwargs["name"] = self.name  # type: ignore
-
-        arguments = self._get_arguments(**kwargs)
-        parameter_names = self._get_parameter_names(arguments)
-        artifact_names = self._get_artifact_names(arguments)
+    use_func_params_in_call: bool = False
```

Comment:
@elliotgunton I wonder if this should be called something simpler like `callable_syntax`?

Reply:
I agree, this was my favourite name from the beginning! https://github.com/argoproj-labs/hera/pull/763#discussion_r1316968243

Reply:
We should use that!